reV/supply_curve/exclusions.py

```diff
@@ -735,7 +735,20 @@ def _generate_ones_mask(self, ds_slice):
         mask : ndarray
             Array of ones slices down by ds_slice
         """
-        mask = np.ones(self.shape, dtype='float32')[ds_slice]
+
+        ones_shape = ()
+        for i, s in enumerate(self.shape):
+            try:
+                ax_slice = ds_slice[i]
+                if np.issubdtype(type(ax_slice), np.integer):
+                    ones_shape += (ax_slice,)
+                else:
+                    ax = np.arange(s, dtype=np.int32)
+                    ones_shape += (len(ax[ax_slice]), )
+            except IndexError:
```

Comment:
I need to test this a bit more, I'm worried that if ax_slice is a list or ndarray with idxs outside the range of ax it will cause and index error... I'll proiritize that tomorrow, mind if I push a commit if needed?

Reply:
definitely. There's already a test framework here you can work on, that's how I fixed some edge cases: `test_supply_curve_exclusions.py::test_no_excl`

Reply:
FYI, as I suspected there was a fringe case, fixed it with and if/else instead of try/except. Will fail now if the ds_slice causes an index error